File: Ruby/civitas/lib/sorpresa_por_casa_hotel.rb
#encoding: utf­8
require_relative "sorpresa.rb"
module Civitas
class SorpresaPorCasaHotel < Sorpresa
  attr_accessor :valor
  def initialize(_valor, _texto)
    super(_texto)
    @valor = _valor
  end
  
  def aplicar_a_jugador(_actual, _todos)
    if(jugador_correcto(_actual, _todos))
      informe(_actual, _todos)
      _todos.at(_actual).modificar_saldo(@valor*_todos.at(_actual).cantidad_casas_hoteles)
    end
  end
  
  def to_ss()
    return @texto
  end
end
end

File: Java/civitas/src/civitas/SorpresaPorCasaHotel.java
package civitas;

import java.util.ArrayList;

public class SorpresaPorCasaHotel extends Sorpresa{
    private int valor;
    
    SorpresaPorCasaHotel(int _valor, String _texto){
        super(_texto);
        valor = _valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).modificarSaldo(this.valor*todos.get(actual).cantidadCasasHoteles());
        }
    }
    
    @Override
    public String toString(){
        return "Sorpresa{ " + super.getTexto() + ", valor " + Integer.toString(valor) + " }";
                 
    }
}
